cmds/basename/basename.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2012-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Basename return name with leading path information removed.
+//
+// Synopsis:
+//     basename NAME [SUFFIX]
+//
+// Options:
+//     -s: optional flag for removing suffix
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+)
+
+var (
+	suffix = flag.String("s", "", "Strip prefix from file")
+)
+
+func init() {
+	// Usage Definition
+	defUsage := flag.Usage
+	flag.Usage = func() {
+		os.Args[0] = "basename [-s] NAME [SUFFIX]"
+		defUsage()
+	}
+}
+
+func main() {
+	flag.Parse()
+	if len(flag.Args()) < 1 {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	if len(*suffix) > 0 {
+		for i := 0; i < len(flag.Args()); i++ {
```

Comment:
I think it would alright to not support multiple NAMEs and remove this branch of the if-statement altogether.